--- /usr/share/maxima/5.24.0/share/utils/max-tex-tools.lisp.orig	2013-08-24 12:45:37.879560593 +0200
+++ /usr/share/maxima/5.24.0/share/utils/max-tex-tools.lisp	2013-08-24 13:03:31.508884428 +0200
@@ -133,22 +133,24 @@
 
 
 
-;;* *********************** *
-;;* Allow alliaseds symbols *
-;;* *********************** *
-
-
-(defun tex-atom (x l r)	;; atoms: note: can we lose by leaving out {}s ?
-  (append l
-	  (list (cond ((numberp x) (texnumformat x))
-		      ((and (symbolp x) (or (get x 'reversealias) (get x 'texword))))
-                      ((stringp x) (tex-string x))
-                      ((mstringp x)
-                       (let ((s (maybe-invert-string-case (symbol-name (stripdollar x)))))
-                         (tex-string (quote-% (if $stringdisp (concatenate 'string "``" s "''") s)))))
-                      ((characterp x) (tex-char x))
-		      (t (tex-stripdollar x))))
-	  r))
+; Giorgio Nucci: This breaks the new format for functions like cos: 
+; \cos\left( right)
+;;;* *********************** *
+;;;* Allow alliaseds symbols *
+;;;* *********************** *
+
+
+;(defun tex-atom (x l r)	;; atoms: note: can we lose by leaving out {}s ?
+  ;(append l
+	  ;(list (cond ((numberp x) (texnumformat x))
+		      ;((and (symbolp x) (or (get x 'reversealias) (get x 'texword))))
+                      ;((stringp x) (tex-string x))
+                      ;((mstringp x)
+                       ;(let ((s (maybe-invert-string-case (symbol-name (stripdollar x)))))
+                         ;(tex-string (quote-% (if $stringdisp (concatenate 'string "``" s "''") s)))))
+                      ;((characterp x) (tex-char x))
+		      ;(t (tex-stripdollar x))))
+	  ;r))
 
 
 ;;* *************** *
